### PR TITLE
aws: support for sts assume role

### PR DESCRIPTION
**Pull request by bonifaido**
Signed-off-by: Nandor Kracser <bonifaido@gmail.com>

| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?

Support for setting for STS Assume role when using AWS credentials.

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [ ] User guide and development docs updated (if needed)

### CODE REVIEW

File: internal/cli/command/controlplane/init.go [hunk 1]
```diff
@@ -153,6 +154,7 @@ func askProvider(k8sContext string) (string, error) {
 }
 
 func runInit(options initOptions, banzaiCli cli.Cli) error {
+	println("itt jarunk")
```
Comment (orymate) [inline internal/cli/command/controlplane/init.go:157]:
```suggestion
	println("itt jarunk")
```

Comment (bonifaido) [reply internal/cli/command/controlplane/init.go:157]: totally draft! :)

Comment (sagikazarmark) [reply internal/cli/command/controlplane/init.go]: 😂

File: internal/cli/input/secret.go [hunk 2]
```diff
@@ -95,17 +97,23 @@ func GetAmazonCredentials(allowSessionToken bool) (string, map[string]string, er
...
+		if err != nil {
+			return "", nil, err
+		}
+		out = map[string]string{
+			"AWS_ACCESS_KEY_ID":     value.AccessKeyID,
+			"AWS_SECRET_ACCESS_KEY": value.SecretAccessKey,
+			"AWS_SESSION_TOKEN":     value.SessionToken,
+		}
+	} else {
+		out = map[string]string{
```
Comment (orymate) [inline internal/cli/input/secret.go:113]: I'd keep the ability to use a session token without assuming a role. (Use case: user runs the cli on a shared jump host, and neither wants to use their original iam credentials, nor to assume a role.)

Comment (bonifaido) [reply internal/cli/input/secret.go:113]: Good point!

**Pull request merged**